src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,756 @@
+/*global optimizeforsource, mappingData, currentDelimiter, currentHeaderlines, preview, addRow, delLastRow,
+ * resetMapping, showstep1, isExcel, getLocations, initDatastream, loadMapping, returnRows*/
 /**
+ * 
  * shows the popup to create a new Thing
  */
 function showThingModal() {
-    var $modal = $("#indexfooter").find("button:eq(0)");
-    $modal.attr("onclick", "createThing()");
-    $modal.html("Create Thing");
-    $modal.show();
-    modal("dialog", "thing.html", getLocations, "Create a Thing");
+	var $modal = $("#indexfooter").find("button:eq(0)");
+	$modal.attr("onclick", "createThing()");
+	$modal.html("Create Thing");
+	$modal.show();
+	modal("dialog", "thing.html", getLocations, "Create a Thing");
 }
 
 /**
  * opens the popup to create a new datastream
  */
 function showStreamModal() {
-    var $modal = $("#indexfooter").find("button:eq(0)");
-    $modal.attr("onclick", "createDS()");
-    $modal.html("Create Datastream");
-    $modal.show();
-    modal("dialog", "datastream.html", initDatastream, "Create a Datastream");
+	var $modal = $("#indexfooter").find("button:eq(0)");
+	$modal.attr("onclick", "createDS()");
+	$modal.html("Create Datastream");
+	$modal.show();
+	modal("dialog", "datastream.html", initDatastream, "Create a Datastream");
 }
 
 /**
  * opens the popup to edit the mapping
  */
 function showMappingModal() {
-    var $modal = $("#indexfooter").find("button:eq(0)");
-    $modal.attr("onclick", "saveMapping()");
-    $modal.html("Save Mapping");
-    $modal.show();
-    modal("dialog", "mapping.html", loadMapping, "Current\xa0Mapping");
+	var $modal = $("#indexfooter").find("button:eq(0)");
+	$modal.attr("onclick", "saveMapping()");
+	$modal.html("Save Mapping");
+	$modal.show();
+	modal("dialog", "mapping.html", loadMapping, "Current\xa0Mapping");
 }
 
 /**
  * opens the popup to show the failed rows
  */
 function showReturnModal() {
-    $("#indexfooter").find("button:eq(0)").hide();
-    modal("dialog", "returnRows.html", returnRows, "Skipped Rows");
+	$("#indexfooter").find("button:eq(0)").hide();
+	modal("dialog", "returnRows.html", returnRows, "Skipped Rows");
+}
+
+/**
+ * adds a message to the log
+ * 
+ * @param msg
+ *            the message
+ */
+function addToLog(msg) {
+	var date = new Date();
+	var time;
+	hours = pad2(date.getHours());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['pad2' was used before it was defined.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)